webdriver-extension/index.bs

```diff
@@ -0,0 +1,553 @@
+<dl>
+  <dt>[=Mock Sensor Reading Values=]</dt>
+  <dd>
+    <pre class="idl">
+      dictionary UncalibratedMagnetometerReadingValues {
+        required double? x;
+        required double? y;
+        required double? z;
+        required double? xBias;
+        required double? yBias;
+        required double? zBias;
+      };
+    </pre>
+  </dd>
+</dl>
+
+### Absolute Orientation Sensor ### {#mock-absolute-orientation-sensor}
+The <dfn for="MockSensorType" enum-value>"absolute-orientation"</dfn> type is the mock sensor type
+associated with the usage of the <a>AbsoluteOrientationSensor</a> interface.
+<dl>
+  <dt>[=Mock Sensor Reading Values=]</dt>
+  <dd>
+    <pre class="idl">
+      dictionary AbsoluteOrientationReadingValues {
+        required FrozenArray&lt;double>? quaternion;
+      };
+    </pre>
+  </dd>
+</dl>
+
+### Relative Orientation Sensor ### {#mock-relative-orientation-sensor}
+The <dfn for="MockSensorType" enum-value>"relative-orientation"</dfn> type is the mock sensor type
+associated with the usage of the <a>RelativeOrientationSensor</a> interface.
+<dl>
+  <dt>[=Mock Sensor Reading Values=]</dt>
+  <dd>
+    <pre class="idl">
+      dictionary RelativeOrientationReadingValues : AbsoluteOrientationReadingValues {
+      };
+    </pre>
+  </dd>
+</dl>
+
+### Geolocation Sensor ### {#mock-geolocation-sensor}
+The <dfn for="MockSensorType" enum-value>"geolocation"</dfn> type is the mock sensor type
+associated with the usage of the <a>GeolocationSensor</a> interface.
+<dl>
+  <dt>[=Mock Sensor Reading Values=]</dt>
+  <dd>
+    <pre class="idl">
+      dictionary GeolocationReadingValues {
+        required double? latitude;
+        required double? longitude;
+        required double? altitude;
+        required double? accuracy;
+        required double? altitudeAccuracy;
+        required double? heading;
+        required double? speed;
+      };
+    </pre>
+  </dd>
+</dl>
+
+### Proximity Sensor ### {#mock-proximity-sensor}
+The <dfn for="MockSensorType" enum-value>"proximity"</dfn> type is the mock sensor type
+associated with the usage of the <a>ProximitySensor</a> interface.
+<dl>
+  <dt>[=Mock Sensor Reading Values=]</dt>
+  <dd>
+    <pre class="idl">
+      dictionary ProximityReadingValues {
+        required double? distance;
+        required double? max;
+        required boolean? near;
+      };
+    </pre>
+  </dd>
+</dl>
+
+Extensions {#webdriver-extensions}
+=====================
+
+## Commands ## {#mock-sensor-commands}
+
+### Create mock sensor ### {#create-mock-sensor-command}
+
+<table>
+  <tbody>
+    <tr>
+      <th>HTTP Method</th>
+      <th><a lt="extension command URI template">URI Template</a></th>
+    </tr>
+    <tr>
+      <td>POST</td>
+      <td>/session/{session id}/sensor</td>
+    </tr>
+  </tbody>
+</table>
+
+The <dfn>create mock sensor</dfn> <a>extension command</a> simulates user
+creation of a [=mock sensor=].
+
+The <a>remote end steps</a> are:
```

Comment:
Sounds like all the extension commands only affect the current browsing context, not the entire session? For such commands, there should be a step "If the current top-level browsing context is no longer open, return error with error code no such window" near the beginning of the remote end steps. It may also worth considering adding "Handle any user prompts and return its value if it is an error."

Reply:
@JohnChen0, thanks for catching this! Seems [Permission API](https://w3c.github.io/permissions/#set-permission) and [Reporting API](https://w3c.github.io/permissions/#set-permission) have similar issue. Are these two steps both necessary for them? Could you provide some guideline on which one I should add and when?

Reply:
Any WebDriver command that targets a specific top-level browsing context (i.e., a tab or a window) should have a step to check if that browsing context is still open. A WebDriver session has a current browsing context. If the tab or window associated with the current browsing context is closed, then the WebDriver implementation can't carry out the command, and should return an error. The implementation doesn't automatically switch to a different browsing context -- it must be changed explicitly by Switch To Window command. Even if the step is missing from the spec, the implementation would likely be forced to check for this error any way, as there is likely no good way to carry out the command with an invalid current browsing context.

Most commands should also handle user prompts, unless there is some specific reason not to do so. When a prompt is shown, browser functionality is severely limited (e.g., JavaScript code can't be run), and may make it difficult to execute the command.

Reply:
If the command primarily affects the current frame (as selected by WebDriver Switch To Frame command), then we should check if the current browsing context is still open. If it primarily affects the current tab/window, then we should check current top-level browsing context.

Reply:
@JohnChen0, thanks again. I will fix it base on @rakuco's [proof-reading](https://github.com/Honry/sensors/pull/2).